### ISSUE TITLE
Master/Stack layout strange behavior

### ISSUE THREAD

**Issue by Mango0x45** [hyprwm/Hyprland#1002]
I've been using the master and stack layout for a while and I've come
across some interesting behavior.  If I open 2 windows in a workspace I
will get a split like so (`X` and `Y` represent the two windows):

```
Workspace 1    Workspace 2

  [ X Y ]        [ - - ]
  [ X Y ]        [ - - ]
```

So far the behavior is correct.  If I then go to a second workspace and
do the same thing, I get the same split there:

```
Workspace 1    Workspace 2

  [ X Y ]        [ X Y ]
  [ X Y ]        [ X Y ]
```

However now if I close window `X` in workspace 2, my windows get totally
screwed up and I am left with this layout:

```
Workspace 1    Workspace 2

  [ X - ]        [ - Y ]
  [ Y - ]        [ - Y ]
```

But the expected splits should be this:

```
Workspace 1    Workspace 2

  [ X Y ]        [ Y Y ]
  [ X Y ]        [ Y Y ]
```

I have also tested this with more workspaces, however it seems that the
only workspaces that get affected are the current workspace where the
master window was closed, and the very next workspace you switch to.

**Comment by Quantenzitrone** [comment]
i have this bug too

**Comment by vaxerski** [comment]
https://user-images.githubusercontent.com/43317083/201527042-c3c02e19-10b9-488d-8a84-dd3b9edb127f.mp4

**Comment by vaxerski** [comment]
fixed in 884fc4f89c21d670a3703388924ebbe1099e7a51
